### PR TITLE
Enable ReplicaSets in e2e tests.

### PR DESCRIPTION
**Pull request by madhusudancs**
This is part of an ongoing work to create ReplicaSet and move Deployments to the new selector format. Only the last commit needs review. This PR is based on other PRs which are in review separately, unfortunately there is no easy way to do diff against PRs.

### CODE REVIEW

File: cluster/common.sh [hunk 1]
```diff
@@ -440,6 +440,15 @@ function build-runtime-config() {
           fi
       fi
   fi
+  if [[ "${ENABLE_REPLICASETS}" == "true" ]]; then
```
Comment (bgrant0607) [inline cluster/common.sh]: We should enable ReplicaSet by default in the code, not via flag.

Comment (bgrant0607) [inline cluster/common.sh]: Add it to the list of enabled resources:
```
 func (m *Master) getExtensionResources(c *Config) map[string]rest.Storage {
 	// All resources except these are disabled by default.
 	enabledResources := sets.NewString("jobs", "horizontalpodautoscalers", "ingresses", "podsecuritypolicy")
```

Comment (madhusudancs) [inline cluster/common.sh]: Done.

**Comment by madhusudancs** [comment]
@bgrant0607 Addressed the comments. PTAL.

**Comment by bgrant0607** [comment]
Lgtm

**Pull request merged**